pkg/cmd/roachprod/main.go

```diff
@@ -1171,6 +1172,45 @@ func validateAndConfigure(cmd *cobra.Command, args []string) {
 	}
 }
 
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "check TeamCity for a new roachprod binary and update if available",
+	Long: "Will attempt to download the latest master branch roachprod binary from teamcity" +
+		" and swap the current roachprod with it. The current roachprod binary will be backed up" +
+		" and can be restored via `roachprod update --revert`.",
+	Run: wrap(func(cmd *cobra.Command, args []string) error {
+		currentBinary, err := os.Executable()
+		if err != nil {
+			return err
+		}
+
+		if revertUpdate {
+			if upgrade.PromptYesNo("Revert to previous version? Note: this will replace the" +
+				" current roachprod binary with a previous roachprod.bak binary.") {
+				if err := upgrade.SwapBinary(currentBinary, currentBinary+".bak"); err != nil {
+					return err
+				}
+				fmt.Println("roachprod successfully reverted, run `roachprod -v` to confirm.")
+			}
+			return nil
+		}
+
+		newBinary := currentBinary + ".new"
+		if err := upgrade.DownloadLatestRoadprod(newBinary); err != nil {
+			return err
+		}
+
+		if upgrade.PromptYesNo("Continue with update? This will overwrite any existing roachprod.bak binary.") {
+			if err := upgrade.SwapBinary(currentBinary, newBinary); err != nil {
+				return errors.WithDetail(err, "unable to update binary")
+			}
+
+			fmt.Println("Update successful: run `roachprod -v` to confirm.")
```

Comment:
For non-engineering folks, it would probably be helpful to run this automatically on their behalf. Also `roachprod -v` might not be a valid command if it's not in `$PATH`.